### PR TITLE
fix: Fix type declaration, upgrade bundler and clean up example

### PR DESCRIPTION
**Pull request by shuding**
This PR changes `index.d.ts` to `index.d.mts` as it's needed when we have `import {} from './mod.mjs'`. There're also a few unused imports and other changes that are not synced from our deployed version.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
       "require": "./dist/index.js"
     },
     "./rsc": {
-      "types": "./rsc/dist/index.d.ts",
+      "types": "./rsc/dist/index.d.mts",
```
Comment (lgrammel) [inline packages/core/package.json:47]: does this work with typescript / commonjs setups?

File: examples/next-ai-rsc/package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "@radix-ui/react-tooltip": "^1.0.7",
     "@vercel/analytics": "^1.2.2",
     "@vercel/kv": "^1.0.1",
-    "ai": "3.0.4",
+    "ai": "latest",
```
Comment (lgrammel) [inline examples/next-ai-rsc/package.json:22]: why is this necessary?

File: packages/core/package.json (hunk 1 above)
Comment (shuding) [reply packages/core/package.json:47]: Since this subpath only has ESM builds, so no CJS related issues. We have a shared .mjs file in dist and the import must be `import ... from './mod.mjs'` because as per the spec, the extension can't be omitted. Then, due to https://www.typescriptlang.org/docs/handbook/modules/reference.html#the-moduleresolution-compiler-option, when TS sees that `./mod.mjs` it will look for `.d.mts` instead of `.d.ts`.

File: examples/next-ai-rsc/package.json (hunk 2 above)
Comment (shuding) [reply examples/next-ai-rsc/package.json:22]: This avoids that we need to bump the version number here manually whenever we do an AI package release.

**Review by lgrammel — APPROVED** [review]

File: examples/next-ai-rsc/package.json (hunk 2 above)
Comment (lgrammel) [reply examples/next-ai-rsc/package.json:22]: shouldn't this happen automatically as it does for all other packages?
